java/ql/lib/semmle/code/java/Member.qll

```diff
@@ -201,34 +208,60 @@ class Callable extends StmtParent, Member, @callable {
    * representation is used. If this callable has no parameters, the result is `()`.
    *
    * For example, the method `void printf(java.lang.String, java.lang.Object...)`
-   * has the params string `(String, Object[])`.
+   * has the params string `(String, Object[])`. For parameters of parameterized type
+   * all their type arguments are included, for example,
+   * `Function<? super String,? extends Integer>` (note that here a comma without space
+   * separates the arguments).
    */
-  pragma[nomagic]
   string paramsString() {
-    exists(int n | n = this.getNumberOfParameters() |
```

Comment:
Have rewritten this to use `concat`. The public Git history did not indicate why this was previously implemented this way. So if there are any internal reasons for this, I can revert the change if you want.